cvat/apps/lambda_manager/views.py

```diff
@@ -0,0 +1,386 @@
+        return self._http(method="post", port=func.port, data=payload)
+
+class LambdaFunction:
+    def __init__(self, gateway, data):
+        # ID of the function (e.g. omz.public.yolo-v3)
+        self.id = data['metadata']['name']
+        # type of the function (e.g. detector, interactor)
+        self.kind = data['metadata']['annotations'].get('type')
+        # dictionary of labels for the function (e.g. car, person)
+        spec = json.loads(data['metadata']['annotations'].get('spec') or '[]')
+        labels = [item['name'] for item in spec]
+        if len(labels) != len(set(labels)):
+            raise ValidationError(
+                "`{}` lambda function has non-unique labels".format(self.id),
+                code=status.HTTP_404_NOT_FOUND)
+        self.labels = labels
+        # state of the function
+        self.state = data['status']['state']
+        # description of the function
+        self.description = data['spec']['description']
+        # http port to access the serverless function
+        self.port = data["status"].get("httpPort")
+        # framework which is used for the function (e.g. tensorflow, openvino)
+        self.framework = data['metadata']['annotations'].get('framework')
+        # display name for the function
+        self.name = data['metadata']['annotations'].get('name', self.id)
+        self.gateway = gateway
+
+    def to_dict(self):
+        response = {
+            'id': self.id,
+            'kind': self.kind,
+            'labels': self.labels,
+            'state': self.state,
+            'description': self.description,
+            'framework': self.framework,
+            'name': self.name,
+        }
+
+        return response
+
+    def invoke(self, db_task, frame, quality, mapping, points=None):
+        payload = {
+            'image': self._get_image(db_task, frame, quality),
+            'points': points,
+        }
+
+        response = self.gateway.invoke(self, payload)
+        if mapping:
+            for item in response:
+                item['label'] = mapping.get(item['label'], item['label'])
+
+        return response
+
+    def _get_image(self, db_task, frame, quality):
+        if quality == "original":
+            quality = FrameProvider.Quality.ORIGINAL
+        # FIXME: if quality is none we should use original quality
+        # but now we have some problems with nuclio.
+        elif quality is None or quality == "compressed":
+            quality = FrameProvider.Quality.COMPRESSED
+        else:
+            raise ValidationError(
+                '`{}` lambda function was run '.format(self.id) +
+                'with wrong arguments (quality={})'.format(quality),
+                code=status.HTTP_400_BAD_REQUEST)
+
+        frame_provider = FrameProvider(db_task.data)
+        image = frame_provider.get_frame(frame, quality=quality)
+        return base64.b64encode(image[0].getvalue()).decode('utf-8')
+
+
+class LambdaQueue:
+    def _get_queue(self):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5729179)